src/backend/distributed/planner/multi_explain.c

```diff
@@ -358,6 +359,9 @@ ExplainTaskList(List *taskList, ExplainState *es)
 	ListCell *remoteExplainCell = NULL;
 	List *remoteExplainList = NIL;
 
+	/* make sure that the output is consistent */
+	taskList = SortList(taskList, CompareTasksByTaskId);
```

Comment:
This is contingent upon the planning and `taskId` assignment being deterministic, but if it is, this is an improvement over the status quo, so I'm fine merging this without further inspection (AFAIK there is not a strong reason to assume task id assignment is ever non-deterministic, but just bringing it up).